### PR TITLE
fix(Gorgone): Add missing centengine report part to centreon-storage database

### PR DESCRIPTION
**Pull request by Evan-Adam**
## Description
some of the poller performance report is missing since the gorgone migration.
This pr aim to add all data that was previously added in centreon-storage.

**Fixes** # MON-37990

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>

on central, launch gorgone with statistic module enabled (default configuration) 
wait 5 to 10 minutes for statistics job to be run (see gorgone log
generate a poller performance report and check all data are present, or check in db  : 
here an exemple with one central and one poller : 
```
MariaDB [centreon-storage]> select * from nagios_stats;
+-------------+----------+------------+------------------------------+
| instance_id | stat_key | stat_value | stat_label                   |
+-------------+----------+------------+------------------------------+
|           1 | Min      | 0.102      | Service Check Latency        |
|           1 | Max      | 0.955      | Service Check Latency        |
|           1 | Average  | 0.550      | Service Check Latency        |
|           1 | Min      | 0.001      | Service Check Execution Time |
|           1 | Max      | 0.332      | Service Check Execution Time |
|           1 | Average  | 0.132      | Service Check Execution Time |
|           1 | Min      | 0.020      | Host Check Latency           |
|           1 | Max      | 0.868      | Host Check Latency           |
|           1 | Average  | 0.475      | Host Check Latency           |
|           1 | Min      | 0.030      | Host Check Execution Time    |
|           1 | Max      | 0.152      | Host Check Execution Time    |
|           1 | Average  | 0.083      | Host Check Execution Time    |
|           2 | Min      | 0.102      | Service Check Latency        |
|           2 | Max      | 0.955      | Service Check Latency        |
|           2 | Average  | 0.550      | Service Check Latency        |
|           2 | Min      | 0.001      | Service Check Execution Time |
|           2 | Max      | 0.332      | Service Check Execution Time |
|           2 | Average  | 0.132      | Service Check Execution Time |
|           2 | Min      | 0.020      | Host Check Latency           |
|           2 | Max      | 0.868      | Host Check Latency           |
|           2 | Average  | 0.475      | Host Check Latency           |
|           2 | Min      | 0.030      | Host Check Execution Time    |
|           2 | Max      | 0.152      | Host Check Execution Time    |
|           2 | Average  | 0.083      | Host Check Execution Time    |
+-------------+----------+------------+------------------------------+

```
#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon-gorgone/tests/robot/resources/LogResearch.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                    found = True
+                    res.append(line)
+                    break
+            if not found:
+                return False, c
+
+        return True, res
+    except IOError:
+        logger.console("The file '{}' does not exist".format(log))
+        return False, content[0]
```
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply centreon-gorgone/tests/robot/resources/LogResearch.py:71]: close() is called next line.

File: centreon-gorgone/gorgone/modules/centreon/statistics/class.pm [hunk 2]
```diff
@@ -39,6 +39,7 @@ my ($connector);
 
 sub new {
     my ($class, %options) = @_;
+    use Data::Dumper;
```
Comment (omercier) [inline centreon-gorgone/gorgone/modules/centreon/statistics/class.pm:42]:
```suggestion
```

Comment (omercier) [reply centreon-gorgone/gorgone/modules/centreon/statistics/class.pm:42]: It does not seem useful. Probably a remain from a debug message.

**Review by omercier — APPROVED** [review]

**Review by sdepassio — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

File: centreon-gorgone/tests/robot/resources/LogResearch.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        date = datetime.now().timestamp() - 1
+    limit = time.time() + timeout
+    c = ""
+    while time.time() < limit:
+        ok, c = ctn_find_in_log(log, date, content, regex)
+        if ok:
+            return True, c
+        time.sleep(5)
+    logger.console(f"Unable to find '{c}' from {date} during {timeout}s")
+    return False
```
[the remark this answers is not in the document]
Comment (SamiSoftB) [reply centreon-gorgone/tests/robot/resources/LogResearch.py:29]: not bad suggestion

**Review by SamiSoftB — APPROVED** [review]

**Comment by kduret** [comment]
Migrated to https://github.com/centreon/centreon-collect/pull/1522

**Pull request closed without merging**